dex/msgjson/types.go

```diff
@@ -841,6 +841,7 @@ type TradeResumption struct {
 // PreimageRequest is the server-originating preimage request payload.
 type PreimageRequest struct {
 	OrderID        Bytes `json:"orderid"`
+	Commitment     Bytes `json:"commit"`
 	CommitChecksum Bytes `json:"csum"`
```

Comment:
Not entirely sure why CommitChecksum is in this payload.

Reply:
Oh derp it's because server should not be asking for preimages until the epoch is closed.  So client should be remembering this `CommitChecksum` and verifying that this orders doesn't end up in an epoch's match proof with a different `csum`, which would indicate server is cheating (e.g. we handed over our preimage so the server could game the shuffle).